web/modules/weather_data/src/Service/WeatherDataService.php

```diff
@@ -118,74 +127,82 @@ public function getCurrentConditions($route) {
       return NULL;
     }
 
-    // Since we're on the right kind of route, pull out the data we need.
-    $wfo = $route->getParameter("wfo");
-    $gridX = $route->getParameter("gridX");
-    $gridY = $route->getParameter("gridY");
-    $location = $route->getParameter("location");
+    // This object is only created once per request, so we can store the
+    // results of the API call in a member variable. If the variable is false,
+    // we haven't fetched data yet so do that. Otherwise, just return what's in
+    // the variable already.
+    if ($this->currentConditions == FALSE) {
```

Comment:
From here onwards, the changes in this file are just related to whitespace from being indented a level deeper.